run_experiment.py

```diff
@@ -122,19 +131,39 @@ def parse_arguments():
     dir_opts.add_argument('-o', '--output_file_name', help='The full path name to use when saving the plot.')
 
     args = parser.parse_args()
+
+    if args.log_dir is None:
+        args.log_dir = args.model_dir + 'losses.log'
+
     return args
 
 
 def main(args):
     """ The primary entry point for carrying out experiments on pretrained models.
     """ 
-    experiment_name = args.experiment
+    formatter = logging.Formatter('%(asctime)s %(levelname)s - %(funcName)s: %(message)s',
+                                  "%H:%M:%S")
+    logger = logging.getLogger(__name__)
+    logger.setLevel(args.log_level.upper())
+    stream = logging.StreamHandler()
+    stream.setLevel(args.log_level.upper())
+    stream.setFormatter(formatter)
+    logger.addHandler(stream)
+
+    experiment_name = args.model_dir
     dataset = read_dataset_from_specs('experiments/{}/specs.json'.format(experiment_name))
```

Comment:
`dataset = get_metadata(os.path.join(RES_DIR, experiment_name))["dataset"]`
if you feel that a function is mroe readable no issues, but defintely use `get_metadata` in that function: don't duplicate code